xds/src/main/java/io/grpc/xds/PriorityLoadBalancer.java

```diff
@@ -59,6 +61,8 @@ final class PriorityLoadBalancer extends LoadBalancer {
 
   // Includes all active and deactivated children. Mutable. New entries are only added from priority
   // 0 up to the selected priority. An entry is only deleted 15 minutes after its deactivation.
+  // Note that because all configuration updates should be atomic, updates to children can happen
```

Comment:
This is incorrect. Updates to children happens within the synchronization context. When this code is running it is _already_ within the synchronization context, so calling into the child can cause the child to call back into the LB policy and modify the map.

Basically, the problem here is a single-threaded mundane "modified while iterating" problem.